src/TriggerBinding/SqlTriggerBinding.cs

```diff
@@ -107,7 +107,7 @@ private string GetUserFunctionId()
             string websiteName = SqlBindingUtilities.GetWebSiteName(this._configuration);
 
             var methodInfo = (MethodInfo)this._parameter.Member;
-            string functionName = $"{methodInfo.DeclaringType.FullName}.{methodInfo.Name}";
+            string functionName = $"{methodInfo.Name}";
```

Comment:
If someone has two functions in their app that share a name this means they'll share a lease table now.

What is special about the scale controller that you can't get the FullName of the type it's declared in?

Reply:
Build will fail if user tries to create different functions with same name in an app. So that's covered. But if the user creates same function name across different apps - with this change, they could end up using the same lease table.
@AmeyaRele said he can only access the function name on the Scale controller [PR](https://msazure.visualstudio.com/One/_git/AAPT-Antares-ScaleController/pullrequest/9035947).

Reply:
You can have multiple methods with the same name being used for functions as long as they : 

1. Are in different classes
2. Have different FunctionName attribute values

(note this is for C# - other languages may not allow this but since C# does it doesn't matter if others do or not)

e.g. 

```csharp
namespace Company.Function
{
    public static class AddProduct
    {
        [FunctionName("AddProduct")]
        public static IActionResult Run(
            [HttpTrigger(AuthorizationLevel.Anonymous, "post", Route = "addproduct")]
            [FromBody] Product prod,
            [Sql("dbo.Products", "SqlConnectionString")] out Product product)
        {
            product = prod;
            return new CreatedResult($"/api/addproduct", product);
        }
    }
}
```

and 

```csharp
namespace Company.Function.Othernamespace
{
    public static class AddProduct
    {
        [FunctionName("AddProduct2")]
        public static IActionResult Run(
            [HttpTrigger(AuthorizationLevel.Anonymous, "post", Route = "addproduct")]
            [FromBody] Product prod,
            [Sql("dbo.Products", "SqlConnectionString")] out Product product)
        {
            product = prod;
            return new CreatedResult($"/api/addproduct", product);
        }
    }
}
```

Reply:
Most of our samples all have the same method name `Run` - and I'm going to guess this is pretty common for users too so would be pretty impactful to make this change.

Reply:
methodInfo.Name gives me AddProduct for the first one and AddProduct2 for the second one. I'll have to test for non dotnet languages and confirm the same.

Reply:
Well that's odd - can you check if it's pulling it from the class name or the FunctionName?

Might be worth asking the core team where this value is populated from. Because for non-C# languages I'm curious how they would even get a "MethodInfo" given that that's specifically for reflection. But I suppose it would make sense if they decided to shim in their own version of it based on other stuff...

If it using the FunctionName then that should be fine though.

Reply:
I couldn't debug and see what's the value returned for methodInfo.DeclaringType.Name on javascript functions (not able to debug even after replacing the local dll in the bundle path), but I did run two different functions under the same app and it created two different leases table so definitely not taking the method name which is same for both the functions.